src/request/derive-address/DeriveAddressApi.js

```diff
@@ -0,0 +1,61 @@
+/* global TopLevelApi */
+/* global DeriveAddress */
+/* global Key */
+/* global KeyStore */
+/* global Nimiq */
+
+class DeriveAddressApi extends TopLevelApi { // eslint-disable-line no-unused-vars
+    /**
+     * @param {DeriveAddressRequest} request
+     */
+    async onRequest(request) {
+        const parsedRequest = await DeriveAddressApi._parseRequest(request);
+        const handler = new DeriveAddress(parsedRequest, this.resolve.bind(this), this.reject.bind(this));
+        handler.run();
+    }
+
+    /**
+     * @param {DeriveAddressRequest} request
+     * @returns {Promise<ParsedDeriveAddressRequest>}
+     * @private
+     */
+    static async _parseRequest(request) {
+        if (!request) {
+            throw new Error('Empty request');
+        }
+
+        if (!request.appName || typeof request.appName !== 'string') {
+            throw new Error('appName is required');
+        }
+
+        if (!request.keyId || typeof request.keyId !== 'string') {
+            throw new Error('keyId is required');
+        }
+
+        // Check that key exists.
+        const keyInfo = await KeyStore.instance.getInfo(request.keyId);
+        if (!keyInfo) {
+            throw new Error('Unknown keyId');
+        }
+
+        if (keyInfo.type === Key.Type.LEGACY) {
+            throw new Error('Cannot derive addresses for single-account wallets');
```

Comment:
Honestly, I would leave the `(legacy)` out here, since _maybe_ single-account wallets could be supported continously (I just read recently that it's a _feature_ of another blockchain). And also, weren't you the one who wanted to deemphasize the use of the word 'legacy'? ;)

Reply:
We have to decide. I'm OK with both, but in the doc it says legacy. :P
Yep, maybe no "legacy" nowhere.